pytorch_lightning/trainer/trainer.py

```diff
@@ -147,7 +147,8 @@ def __init__(
         reload_dataloaders_every_epoch: bool = False,
         auto_lr_find: Union[bool, str] = False,
         replace_sampler_ddp: bool = True,
-        terminate_on_nan: bool = False,
+        terminate_on_nan: Optional[bool] = None,
```

Comment:
this argument shall go to the end as it won't break compatibility with users using only positional arguments...